### PR TITLE
Feature controller specs

### PR DESCRIPTION
**Pull request by IsmailChangezi**
## Description
- Add Specs For Controllers

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /users/:user_id/posts/:id' do
+    it 'works! (now write some real specs)' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(200)
+      expect(response).to render_template('posts/show')
+      expect(response.body).to include('Post 1 from User 1')
+    end
+  end
+end
```
Comment (prolajumokeoni) [inline spec/requests/posts_spec.rb]: - [ ] Please ensure that index and show actions for  Post controllers are tested, you can create two blocks for index and show and add your tests inside 👍🏿 
````ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
````

- Make sure that for each action you check:
If response status was correct.
If a correct template was rendered.
If the response body includes correct placeholder text.

File: spec/requests/users_spec.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /users/:user_id/posts/:id' do
+    it 'works! (now write some real specs)' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(200)
+      expect(response).to render_template('posts/show')
+      expect(response.body).to include('Post 1 from User 1')
+    end
+  end
+end
```
Comment (prolajumokeoni) [inline spec/requests/users_spec.rb]: - [ ] Please ensure that index and show actions for Users controllers are tested, you can create two blocks for index and show and add your tests inside 👍🏿 
````ruby
describe 'GET /index' do
  ...
end

describe 'GET /show' do
  ...
end
````

- Make sure that for each action you check:
If response status was correct.
If a correct template was rendered.
If the response body includes correct placeholder text.

**Review by prolajumokeoni — CHANGES_REQUESTED** [review]
Hi @IsmailChangezi 
Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️   Added Rspec gem to project 



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by iambenkis — APPROVED** [review]
# Status: APPROVED 🟢 

Hi @IsmailChangezi  ✋

Your project is complete!  There is nothing else to say other than... it's time to merge it: ship it:
Congratulations! :tada: :clap: :clap: :100:

## Highlights!
- Examples  are passing ✔️
- Good commit messages ✔️
- Descriptive Pull request ✔️
- Professional readme ✔️

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me @iambenkis if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**